src/shared-components/immersiveModal/index.tsx

```diff
@@ -211,6 +198,19 @@ export const ImmersiveModal: ImmersiveModal = ({
     const modalDesktopScrollingElementInstance =
       modalDesktopScrollingElement.current;
 
+    const handleScroll = throttle(() => {
```

Comment:
Moving this inside the effect removes the need to specify `handleScroll` as a dependency.